frontend/src/app/map-view/scenario/scenario-area-detail/scenario-area-detail.component.ts

```diff
@@ -139,7 +141,9 @@ export class ScenarioAreaDetailComponent implements OnInit, OnDestroy {
           { areaId: this.areaCoastMatrices.areaMatrices[0].areaId }));
       }
     } else {
-      this.store.dispatch(ScenarioActions.excludeActiveAreaCoastal({ areaId: null }));
+      this.store.dispatch(ScenarioActions.excludeActiveAreaCoastal(
+        { areaId: this.associatedCoastalArea ? -1 : null })
```

Comment:
Does this do anything in the end since both null and -1 are resulting in a truthy includeCoast?

Reply:
You're right, it's a little weird, it would be better to enclose both these branches with a check for this.associatedCoastalArea and always set -1.
If there is no `associatedCoastalArea` then the method would never be called in the first place (it becomes disabled by the template). But I prefer the strictness of checking anyway. Let me fix this.